File: mobile/lib/features/discovery/domain/accommodation_model.dart
class Accommodation {
  final String id;
  final String name;
  final String slug;
  final String type; // hostel, pg, coliving
  final String gender; // boys, girls, coed
  final String address;
  final List<String> images;
  final List<String> amenities;
  final double startingPrice;
  final double? deposit;
  final String verificationStatus;
  final String? description;
  final Map<String, dynamic>? ownerContact;
  final List<dynamic>? nearbyColleges;

  Accommodation({
    required this.id,
    required this.name,
    required this.slug,
    required this.type,
    required this.gender,
    required this.address,
    required this.images,
    required this.amenities,
    required this.startingPrice,
    this.deposit,
    required this.verificationStatus,
    this.description,
    this.ownerContact,
    this.nearbyColleges,
  });

  factory Accommodation.fromJson(Map<String, dynamic> json) {
    final pricing = json['pricing'] ?? {};
    final verification = json['verification'] ?? {};
    
    return Accommodation(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      slug: json['slug'] ?? '',
      type: json['type'] ?? 'hostel',
      gender: json['gender'] ?? 'coed',
      address: json['address'] ?? '',
      images: List<String>.from(json['images'] ?? []),
      amenities: List<String>.from(json['amenities'] ?? []),
      startingPrice: (pricing['startingPrice'] ?? 0).toDouble(),
      deposit: pricing['deposit'] != null ? (pricing['deposit']).toDouble() : null,
      verificationStatus: verification['status'] ?? 'unverified',
      description: json['description'],
      ownerContact: json['ownerContact'],
      nearbyColleges: json['nearbyColleges'],
    );
  }
}

File: backend/src/modules/accommodations/accommodation.model.js
const mongoose = require('mongoose');
const slugify = require('slugify');

const accommodationSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
  },
  slug: {
    type: String,
    unique: true,
    index: true,
  },
  type: {
    type: String,
    enum: ['hostel', 'pg', 'coliving', 'flat'],
    required: true,
  },
  gender: {
    type: String,
    enum: ['boys', 'girls', 'coed'],
    required: true,
  },
  location: {
    type: {
      type: String,
      enum: ['Point'],
      default: 'Point',
    },
    coordinates: {
      type: [Number],
      required: true,
    }
  },
  address: {
    type: String,
    required: true,
  },
  nearbyColleges: [{
    collegeId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'College'
    },
    distanceMeters: Number,
  }],
  verification: {
    status: {
      type: String,
      enum: ['unverified', 'verified', 'premium_verified'],
      default: 'unverified',
    },
    verifiedAt: Date,
    verifiedBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
    }
  },
  images: [String],
  amenities: [String],
  pricing: {
    startingPrice: {
      type: Number,
      required: true,
    },
    deposit: Number,
  },
  ownerId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
  },
  ownerContact: {
    phone: String,
    email: String,
    name: String,
  },
  status: {
    type: String,
    enum: ['active', 'hidden'],
    default: 'active',
  }
}, {
  timestamps: true,
});

// Explicit Indexes
accommodationSchema.index({ location: '2dsphere' });
accommodationSchema.index({ 'pricing.startingPrice': 1 }); // For price sorting/filtering
accommodationSchema.index({ 'nearbyColleges.collegeId': 1, 'verification.status': -1 }); // College-based search prioritizing premium

accommodationSchema.pre('save', function () {
  if (this.isModified('name')) {
    this.slug = slugify(this.name + '-' + Date.now().toString().slice(-4), { lower: true, strict: true });
  }
});

module.exports = mongoose.model('Accommodation', accommodationSchema);
